pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -32,19 +32,34 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(env model.Environmen
 	node model.Proxy) ([]*xdsapi.Listener, error) {
 	config := env.IstioConfigStore
 
-	// collect workload labels
-	workloadInstances, err := env.GetProxyServiceInstances(node)
-	if err != nil {
-		log.Errora("Failed to get gateway instances for router ", node.ID, err)
-		return nil, err
-	}
+	var gateways []model.Config
 
-	var workloadLabels model.LabelsCollection
-	for _, w := range workloadInstances {
-		workloadLabels = append(workloadLabels, w.Labels)
-	}
+	if node.Type == model.Ingress {
```

Comment:
this also needs to go..

Reply:
because the ingress gateway is just another gateway and model.Gateways() already has the ingress gateway with the workload labels of ingress pod.

Reply:
+1